Core/Core/Serialisation/BaseObjectSerializerV2.cs

```diff
@@ -65,7 +66,19 @@ public string Serialize(Base baseObj)
     {
       _stopwatch.Start();
       _isBusy = true;
-      IDictionary<string, object?> converted = PreserializeBase(baseObj, true)!;
+      IDictionary<string, object?> converted;
```

Comment:
Would SDK consumers need to catch ArgumentException? This is one of those times when I wonder if a SpeckleArgumentException might be more easily handled?

Reply:
Partially agree, I think I should switch this to a `SpeckleException`. `ArgumentException` is not an appropriate exception here for the same reason why I don't think `SpeckleArgumentException` is a good idea.

---

`ArgumentException` implies an argument value is not valid for the method call. These validation rules should be consistent and "Functionally pure", i.e. the same input always gives the same result (exception throw or not) regardless of external (runtime) state.

The caller should be-able to easily avoid runtime `ArgumentException` by ensuring the arguments they pass are valid. 
Either by using pre-call checks, or by handling the `ArgumentException` separately.

In this case, we are throwing `ArgumentException` reactively, and its not clear how the caller could avoid this exception if they wanted to.

---

In general I think `ArgumentException` is perfectly valid for us to throw (as long as we don't misuse it). It fits with .NET standards and Microsoft guidance.
If we use `SpeckleArgumentException`, what would this inherit from?
 - If its `ArgumentException`, then what advantage does this give, will we ever want to handle `SpeckleArgumentException` differently from `ArgumentException` I don't think so, and if not, then what value does it add?
- If its `SpeckleException`, we have the opposite question, won't we always want to handle them differently? likely, a developer wants to know if they are responsible for passing around invalid objects or breaking call contracts.  Having `SpeckleArgumentException` inherit from `SpeckleException` introduces a risk that this goes unnoticed, and accidently handled in the same way other general runtime failures (where the caller likely doesn't have control).
We'd be mixing exceptions that are always the caller's fault, with exceptions that are the fault of some external state/service.
- What about `ArgumentNullException` do we replicate this also, leading to same questions as the above (and potential confusion on the callers side)

Reply:
Well we're stating we can throw it, I do feel it's a pain to have to catch multiple types when calling into an API. But happy for you to resolve it.

Reply:
We seem on the same page now, I've updated this to a `SpeckleException`